addon/templates/components/polaris-button.hbs

```diff
@@ -1,63 +1,80 @@
-{{#component
-  buttonComponentName
-  class=class
-  url=url
-  external=external
-  primary=primary
-  destructive=destructive
-  disabled=isDisabled
-  loading=loading
-  size=size
-  outline=outline
-  fullWidth=fullWidth
-  submit=submit
-  plain=plain
-  accessibilityLabel=accessibilityLabel
-  ariaControls=ariaControls
-  ariaExpanded=ariaExpandedValue
-  iconOnly=iconOnly
-  onClick=onClick
-  onFocus=onFocus
-  onBlur=onBlur
-  mouseUp=handleMouseUpByBlurring
-  data-test-id=dataTestId
+{{#with
+  (component "polaris-button/content"
+    text=text
+    primary=primary
+    destructive=destructive
+    loading=loading
+    disclosure=disclosure
+    icon=icon
+  )
+  as |contentComponent|
 }}
-  <span class="Polaris-Button__Content">
-    {{#if loading}}
-      <span class="Polaris-Button__Spinner">
-        {{polaris-spinner
-          size="small"
-          accessibilityLabel="Loading"
-          color=spinnerColor
-        }}
-      </span>
-    {{/if}}
-
-    {{#if icon}}
-      <span class="Polaris-Button__Icon">
-        {{#if (is-component-definition icon)}}
-          {{render-content icon}}
+  {{#if url}}
+    {{#if isDisabled}}
+      {{!--
+        Render an `<a>` so toggling disabled/enabled state changes only the
+        `href` attribute instead of replacing the whole element.
+      --}}
+      <a id={{id}} class={{classes}} aria-label={{accessibilityLabel}} data-test-id={{dataTestId}}>
+        {{#if hasBlock}}
```

Comment:
Does

```handlebars
{{#component contentComponent}}
  {{yield}}
{{/component contentComponent}}
```

do the same thing as

```handlebars
{{component contentComponent}}
```

if there isn't a block to yield? If so maybe we don't need the `hasBlock` check.

Reply:
I'd imagine most of the time, yes, but if `contentComponent` has different behaviour depending on whether it's in block or inline mode then we need to render it in either block or inline mode as appropriate. This is the case with quite a number of the components in `ember-polaris`, which yield if they're used in block mode, or render their `text` attribute otherwise.

I made a [twiddle](https://ember-twiddle.com/3fa614c4930f5c22faccecad45f9c112?openFiles=templates.application.hbs%2C) to show what I mean: both instance of `outer-component` cause `inner-component` to be rendered in inline mode - even if `outer-component` is rendered in block mode. It's a nuisance as your suggested simplification would be very nice, but unfortunately has breaking behaviour (which IIRC we've hit in the early days of `ember-polaris`) 😞